zstacklib/zstacklib/utils/objgraph.py

```
@@ -0,0 +1,960 @@
+
+    .. versionchanged:: 1.7
+       New parameter: ``backrefs``.
+
+    .. versionchanged:: 2.0
+       New parameter: ``output``.
+
+    """
+    backrefs = kw.pop('backrefs', True)
+    chains = [chain for chain in chains if chain]  # remove empty ones
+
+    def in_chains(x, ids=set(map(id, itertools.chain(*chains)))):
```

Ruff (0.8.2)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)